plugins/plugin-chart-table/src/TableChart.tsx

```diff
@@ -57,121 +123,131 @@ export default function TableChart(props: TableChartTransformedProps) {
     showCellBars = true,
     emitFilter = false,
     sortDesc = false,
-    onChangeFilter = () => {},
-    filters: initialFilters = {},
+    onChangeFilter,
+    filters: initialFilters,
+    sticky = true, // whether to use sticky header
   } = props;
 
   const [filters, setFilters] = useState(initialFilters);
 
-  function getValueRange(key: string) {
-    let maxValue;
-    let minValue;
-    if (typeof data?.[0]?.[key] === 'number') {
-      const nums = data.map(row => row[key]) as number[];
-      if (alignPositiveNegative) {
-        maxValue = Math.max(...nums.map(Math.abs));
-        minValue = 0;
-      } else {
-        maxValue = Math.max(...nums);
-        minValue = Math.min(...nums);
+  // only take relevant page size options
+  const pageSizeOptions = useMemo(
+    () => PAGE_SIZE_OPTIONS.filter(([n, _]) => n <= 2 * data.length),
+    [data.length],
+  );
+
+  const columns = useMemo(() => {
+    function getValueRange(key: string) {
```

Comment:
* if you import `d3-array` you can use d3's `max` instead of extent when you don't really need min value. This will also eliminate declaring variables.

```ts
import { max as d3Max, extent as d3Extent } from 'd3-array';

return (alignPositiveNegative
  ? [0, d3Max(nums.map(Math.abs))]
  : d3Extent(nums)) as ValueRange;
```

* perhaps can split `useMemo` into smaller block

```ts
const getValueRange = useMemo(() => (key: string) => {
  ...
}, [data, alignPositiveNegative]);

const isActiveFilterValue = useMemo(() => (key: string, val: DataRecordValue) => {
   ...
}, [filters]);

Reply:
nice